crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -178,6 +220,102 @@ fn create_diagnostic(
+        ));
+    }
+
+    let property_ident = get_binding_identifier(property).unwrap();
+    let mut references = ctx
+        .semantic()
+        .symbol_references(property_ident.symbol_id.get().expect("symbol should be set"));
+
+    let has_spread_attribute = references.any(|x| used_with_spread_attribute(x.node_id(), ctx));
+
+    if has_spread_attribute {
+        // If the JSXElement has a spread attribute, we cannot apply a fix safely,
+        // as the same property name could be exist within the spread attribute.
+        return ctx.diagnostic(only_used_in_recursion_diagnostic(
+            property.span(),
+            get_property_name(property).unwrap(),
+        ));
+    }
+
+    ctx.diagnostic_with_dangerous_fix(
+        only_used_in_recursion_diagnostic(property.span, get_property_name(property).unwrap()),
+        |fixer| {
+            let mut fix = fixer.new_fix_with_capacity(references.count() + 1);
+
+            let source = ctx.source_text();
+            let span_start = skip_to_next_char(source, property.span.start, &Direction::Backward)
+                .unwrap_or(property.span.start);
+            let span_end =
+                skip_to_next_char(ctx.source_text(), property.span.end, &Direction::Forward)
+                    .unwrap_or(property.span.end);
+
+            fix.push(Fix::delete(Span::new(span_start, span_end)));
+            let property_id = get_binding_identifier(property).expect("property_id should be set");
+
+            // search for references to the function and remove the property
+            for reference in ctx
+                .semantic()
+                .symbol_references(property_id.symbol_id.get().expect("symbol should be set"))
+            {
+                let mut ancestors = ctx.nodes().ancestors(reference.node_id());
+
+                let Some(attr) =
+                    ancestors.find_map(|node| match ctx.nodes().get_node(node).kind() {
+                        AstKind::JSXAttributeItem(attr) => Some(attr),
+                        _ => None,
+                    })
+                else {
+                    continue;
+                };
+
+                fix.push(Fix::delete(attr.span()));
+            }
+
+            fix
+        },
+    );
+}
+
+fn used_with_spread_attribute(node_id: NodeId, ctx: &LintContext) -> bool {
+    let opening_element = ctx
+        .nodes()
+        .ancestors(node_id)
+        .find_map(|node| match ctx.nodes().kind(node) {
+            AstKind::JSXOpeningElement(opening_element) => Some(opening_element),
+            _ => None,
+        })
+        .expect("should be safe");
+
+    opening_element
+        .attributes
+        .iter()
+        .any(|attr| matches!(attr, JSXAttributeItem::SpreadAttribute(_)))
+}
+
+fn get_property_name<'a>(property: &'a BindingProperty<'a>) -> Option<&'a str> {
+    match &property.key {
+        PropertyKey::StaticIdentifier(ident) => Some(ident.name.as_str()),
+        _ => None,
+    }
+}
```

Comment:
use `property.key.static_name()` ?

Reply:
Resolved in: 81780ca214d2835e182ecf2257ab99b09fd94587...5e4767a4c15157e3cc36cfe1c9789224c76c6b2c